### PR TITLE
JAMES-3012 Part-1 Moving some common helpers in jmap integration test to a generic maven module

### PR DESCRIPTION
**Pull request by trantienduchn**
That's james-server-testing. In order to reuse them in webadmin integration test

### CODE REVIEW

File: server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/ReindexingWithEventDeadLettersTest.java [hunk 1]
```diff
@@ -100,19 +102,21 @@ void setUp(GuiceJamesServer jamesServer) throws Exception {
...
         webAdminApi = WebAdminUtils.spec(jamesServer.getProbe(WebAdminGuiceProbe.class).getWebAdminPort());
 
-        aliceAccessToken = authenticateJamesUser(baseUri(jamesServer), ALICE, ALICE_PASSWORD);
+        aliceAccessToken = authenticateJamesUser(LocalHostURIBuilder.baseUri(jmapPort), ALICE, ALICE_PASSWORD);
 
         dockerElasticSearch.getDockerES().pause();
         Thread.sleep(Duration.TEN_SECONDS.getValueInMS()); // Docker pause is asynchronous and we found no way to poll for it
     }
 
+    @Disabled("It's already fails for a long time, but CI didn't detect this when it's not marked as BasicFeature")
```
Comment (chibenwa) [inline server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/ReindexingWithEventDeadLettersTest.java]: Jira?

File: server/protocols/webadmin-integration-test/pom.xml [hunk 2]
```diff
@@ -155,5 +184,10 @@
             <artifactId>testcontainers</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.james</groupId>
+            <artifactId>james-server-jmap-draft-integration-testing</artifactId>
+            <scope>test</scope>
```
Comment (chibenwa) [inline server/protocols/webadmin-integration-test/pom.xml]: Is this dependency still needed?

(It should not!)

File: server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/ReindexingWithEventDeadLettersTest.java (hunk 1 above)
Comment (trantienduchn) [reply server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/ReindexingWithEventDeadLettersTest.java]: TODO: add jira JAMES-3011

File: server/protocols/webadmin-integration-test/pom.xml (hunk 2 above)
Comment (trantienduchn) [reply server/protocols/webadmin-integration-test/pom.xml]: one of my attempt to run LinshareIntegrationTest on Memory James but didn't success. It should be removed

**Comment by trantienduchn** [comment]
> but I would really like a webadmin-integration-test-memory ...

With a structure like jmap integration testing (common, memory, cassandra ....) ? I want to go with a simple one, just a maven module with memory James.

**Comment by chibenwa** [comment]
```

[77037d6e8ed4cea3becea0c2fb52d42a7f167058] [ERROR] Errors: 
[77037d6e8ed4cea3becea0c2fb52d42a7f167058] [ERROR]   MemoryFilterTest>FilterTest.messageShouldBeAppendedInInboxWhenCcDoesNotMatchRule:1192 » ConditionTimeout
[77037d6e8ed4cea3becea0c2fb52d42a7f167058] [ERROR]   MemoryFilterTest>FilterTest.messageShouldBeAppendedInInboxWhenRecipientDoesNotMatchRule:1248 » ConditionTimeout
[77037d6e8ed4cea3becea0c2fb52d42a7f167058] [ERROR]   MemoryFilterTest>FilterTest.messageShouldBeAppendedInSpecificMailboxWhenCcRuleMatches:858 » ConditionTimeout
[77037d6e8ed4cea3becea0c2fb52d42a7f167058] [ERROR]   MemoryFilterTest>FilterTest.messageShouldBeAppendedInSpecificMailboxWhenRecipientRuleMatchesCc:914 » ConditionTimeout
[77037d6e8ed4cea3becea0c2fb52d42a7f167058] [ERROR]   MemoryFilterTest>FilterTest.messageShouldBeAppendedInSpecificMailboxWhenToRuleMatches:802 » ConditionTimeout
```

**Comment by trantienduchn** [comment]
:green_apple:

File: server/testing/pom.xml [hunk 3]
```diff
@@ -43,6 +43,10 @@
             <groupId>${james.groupId}</groupId>
             <artifactId>testing-base</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.jayway.jsonpath</groupId>
```
Comment (Arsnael) [inline server/testing/pom.xml]: wrong order

**Comment by chibenwa** [comment]
```
[219e2ec4e93b0dde73f2398e4401398d2fcf2e30] Location: /org/apache/james/imap/scripts/ValidateDeployment.test:499
[219e2ec4e93b0dde73f2398e4401398d2fcf2e30] LastClientMsg: c SEARCH FLAGGED
[219e2ec4e93b0dde73f2398e4401398d2fcf2e30] Expected: '\* SEARCH 1 2 3 4'
[219e2ec4e93b0dde73f2398e4401398d2fcf2e30] Actual   : '* SEARCH'
```

Can be considered :green_apple: but test this please anyway

**Comment by chibenwa** [comment]
I created a separated JIRA issue for this: https://issues.apache.org/jira/browse/JAMES-3013

**Comment by trantienduchn** [comment]
```
[219e2ec4e93b0dde73f2398e4401398d2fcf2e30] Location: /org/apache/james/imap/scripts/ValidateDeployment.test:499
[219e2ec4e93b0dde73f2398e4401398d2fcf2e30] LastClientMsg: c SEARCH FLAGGED
[219e2ec4e93b0dde73f2398e4401398d2fcf2e30] Expected: '\* SEARCH 1 2 3 4'
[219e2ec4e93b0dde73f2398e4401398d2fcf2e30] Actual   : '* SEARCH'
```
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**